named_arrays/_matrices/matrices.py

```diff
@@ -62,30 +72,98 @@ def is_square(self):
         Check if this matrix has the same number of rows and columns.
         """
         if self.is_consistent:
-            rows = self.rows
+            rows = self.cartesian_nd.rows
             num_rows = len(rows)
             return all(num_rows == len(rows[r].components) for r in rows)
         else:
             return False
 
+    @property
+    def vector(self) -> na.AbstractExplicitVectorArray:
+        new_dict = {}
+        for c in self.components:
+            component = self.components[c]
+            if isinstance(component, AbstractMatrixArray):
+                new_dict[c] = component.vector
+            else:
+                new_dict[c] = component
+        return self.type_vector.from_components(new_dict)
```

Comment:
Looking at this again, so if this is called on 2d matrix of 2d vectors, does it become a 2d vector of 2d vectors?